src/Middleware/HeaderPropagation/src/DependencyInjection/HeaderPropagationHttpClientBuilderExtensions.cs

```diff
@@ -22,7 +23,48 @@ public static IHttpClientBuilder AddHeaderPropagation(this IHttpClientBuilder bu
 
             builder.Services.AddHeaderPropagation();
 
-            builder.AddHttpMessageHandler<HeaderPropagationMessageHandler>();
+            builder.AddHttpMessageHandler(services =>
+            {
+                var options = new HeaderPropagationClientOptions();
+                var middlewareOptions = services.GetRequiredService<IOptions<HeaderPropagationOptions>>();
+                for (int i = 0; i < middlewareOptions.Value.Headers.Count; i++)
+                {
+                    var header = middlewareOptions.Value.Headers[i];
+                    options.Headers.Add(header.CapturedHeaderName, header.CapturedHeaderName);
+                }
+                return new HeaderPropagationMessageHandler(Options.Options.Create(options), services.GetRequiredService<HeaderPropagationValues>());
+            });
+
+            return builder;
```

Comment:
What you want to do here is base this on named options using `builder.Name` as the key. https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.2#named-options-support-with-iconfigurenamedoptions